src/composites/TreeDisplay/nodeDescriptionPanel/NodeDescriptionPanel.tsx

```diff
@@ -2,34 +2,75 @@ import "./NodeDescriptionPanel.css"
 import {useEffect, useState} from "react";
 import {determineNodeInfo} from "./NodeDescriptionPanelHelpers";
 
-export default function NodeDescriptionPanel(props) {
+export default function NodeDescriptionPanel(props: { nodes: any[]; impacts: any }) {
+  const [nodes, setNodes] = useState<any[]>([]);
+  const [orderBy, setOrderBy] = useState<string>("default");
+  const [orderDirection, setOrderDirection] = useState<string>("asc");
 
-    const [nodes,setNodes] = useState([]);
+  // Rerender this component whenever the nodes prop from TreeDisplay changes.
+  useEffect(() => {
+    setNodes(props.nodes);
+  }, [props.nodes]);
```

Comment:
Why do we need this useEffect? The component should automatically re-render when props change. Also, using setState inside useEffect can cause infinte render loops.

Reply:
removed this useeffect